src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemOrchestrator.cs

```diff
@@ -78,6 +88,35 @@ await StoreWorkitemBlobAsync(identifier, dataset, cancellationToken)
             }
         }
 
+        /// <inheritdoc />
```

Comment:
Sorry commenting here for AddWorkitemAsync(). What happens if IndexStore call passes and StoreBlob fails and TryDelete fails? Do we have just the partially created work item in a bad state?

Reply:
Added it in a separate PR